ansible/roles/ocp-client-vm/tasks/packages.yml

```diff
@@ -16,36 +16,49 @@
   - cri-o
   - cri-tools
   - podman
+  - tree
   tags:
   - install_openshift_client_vm_packages
 
+
+- name: gather existing maven version {{maven_version}}
+  stat: 
+    path: "/usr/local/apache-maven-{{maven_version}}/bin/mvn"
+  register: existing_mvn_version
+  ignore_errors: true
+  changed_when: false
+
 - name: Get recent version of maven
   get_url:
-    url: https://www-us.apache.org/dist/maven/maven-3/3.5.4/binaries/apache-maven-3.5.4-bin.tar.gz
+    url: https://www-us.apache.org/dist/maven/maven-3/{{maven_version}}/binaries/apache-maven-{{maven_version}}-bin.tar.gz
     dest: /root/apache-maven.tar.gz
   register: r_geturl
   retries: 5
   delay: 20
   until: r_geturl is succeeded
   tags:
   - install_openshift_client_vm_packages
+  when: 'existing_mvn_version.stat.exists == False'
```

Comment:
@jbride  I'd prefer a block around the "Maven Install". That way the when condition is only evaluated once. See the section "Install everything for student user" as an example.